littleblog/entry/views.py

```diff
@@ -20,37 +23,17 @@ class EntryViewSet(viewsets.ModelViewSet):
     queryset = Entry.objects.all()
     permission_classes = [IsAuthenticated, IsOwnerOrReadOnly]
 
+    @action(detail=True, methods=['patch'])
+    def rate(self, request, pk=None):
+        instance = self.get_object()
+        serializer = RatingUpdateSerializer(instance, data=request.data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            set_mark(serializer.instance, **serializer.validated_data)  # rating logic storing in services.py
+            serializer.save()
+            return Response(serializer.data)
+        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
     def get_serializer_class(self):
         if self.action == 'create':
             return EntryCreateSerializer
         return EntryDetailSerializer
-
-
-class RateUpdateView(generics.UpdateAPIView):
-    """
-    Вот это есть та самая вьюшка, которая отвечает за подсчет рейтинга записи.
-    В url нужно дописать в виде параметров pk и mark (оценка, которую ставишь) и отправить PATCH request
-    """
-
-    permission_classes = (IsAuthenticated,)
-    serializer_class = RateUpdateSerializer
-
-    def update(self, request, *args, **kwargs):
-        instance = self.get_object()
-        instance.sum_of_marks += int(request.query_params['mark'])
-        instance.amount_of_marks += 1
-        instance.current_rate = instance.sum_of_marks / instance.amount_of_marks
-        instance.save()
-
-        if getattr(instance, '_prefetched_objects_cache', None):
-            instance._prefetched_objects_cache = {}
-
-        return Response(self.serializer_class(instance).data)
-
-    def get_queryset(self):
-        return Entry.objects.all()
-
-    def get_object(self):
-        queryset = self.filter_queryset(self.get_queryset())
-        obj = generics.get_object_or_404(queryset, pk=self.request.query_params['pk'])
-        return obj
```

Comment:
🔥 👍🏼 

The new implementation looks much more elegant, yeah?

Reply:
Yeah, I like it's new look : D